core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -49,6 +49,16 @@ public class PartitionsTable extends BaseMetadataTable {
         new Schema(
             Types.NestedField.required(1, "partition", Partitioning.partitionType(table)),
             Types.NestedField.required(4, "spec_id", Types.IntegerType.get()),
+            Types.NestedField.optional(
```

Comment:
Sorry we may need to change some test, but do you think we can add it at the end?  I know we sometimes add in the middle if it makes sense, but there's always the small risk some user is selecting some columns by position, so I would say we would put it in the middle only if there's a logical reason.  Or did you put it here for some logical reason?

Reply:
yeah that make sense, especially those 2 columns are optional now.